co2calculator/calculate.py

```diff
@@ -178,163 +252,100 @@ def apply_detour(distance: Kilometer, transportation_mode: str) -> Kilometer:
     return distance_with_detour
 
 
-def calc_co2_bus(
-    distance: Kilometer = None,
-    stops: list = None,
-    size: str = None,
-    fuel_type: str = None,
-    occupancy: int = None,
-    vehicle_range: str = None,
-) -> Tuple[Kilogram, Kilometer]:
+class Bus(BaseModel):
+    """Vehicle type Bus"""
+
+    @enum.unique
+    class Size(str, enum.Enum):
+        """Sizes for busses"""
+
+        MEDIUM = "medium"
+        LARGE = "large"
+        AVERAGE = "average"
+
+    class FuelType(str, enum.Enum):
+        """Fuel types for busses"""
+
+        DIESEL = "diesel"
+
+    class BusRange(str, enum.Enum):
+        """Range types for busses"""
+
+        LONG_DISTANCE = "long-distance"
+
+    size: Size
+    fuel_type: FuelType
+    occupancy: int  # TODO: Validate to 20, 50, 80 or 100
+    vehicle_range: BusRange
+
+
+class BusEmissionRequest(EmissionRequest):
+    """Request model for bus emission calculation"""
+
+    vehicle: Bus
+
+
+def calc_co2_bus(request: BusEmissionRequest) -> float:
     """
     Function to compute the emissions of a bus trip.
-    :param distance: Distance travelled in km;
-                        alternatively param <stops> can be provided
-    :param stops: List of locations as dictionaries in the form
-                        e.g.,  [{"address": "Im Neuenheimer Feld 348",
-                                "locality": "Heidelberg",
-                                 "country": "Germany"},
-                                 {"country": "Germany",
-                                 "locality": "Berlin",
-                                 "address": "Alexanderplatz 1"}]
-                        can have intermediate stops (multiple dictionaries within the list)
-                        alternatively param <distance> can be provided
-    :param size: size class of the bus;                 ["medium", "large", "average"]
-    :param fuel_type: type of fuel the bus is using;    ["diesel"]
-    :param occupancy: number of people on the bus       [20, 50, 80, 100]
-    :param vehicle_range: range/haul of the vehicle     ["local", "long-distance"]
-    :type distance: float
-    :type stops: list[*dict]
-    :type size: str
-    :type fuel_type: str
-    :type occupancy: int
-    :type vehicle_range: str
-    :return: Total emissions of trip in co2 equivalents, distance of the trip
-    :rtype: tuple[float, float]
+    :param request: ...
+    :return: Total emissions of trip in co2 equivalents
     """
-    # NOTE: vehicle_rage 'local' fails with IndexError
-
-    transport_mode = "bus"
 
-    # Set default values
-    if size is None:
-        size = "average"
-        warnings.warn(f"Size of bus was not provided. Using default value: '{size}'")
-    if fuel_type is None:
```

Comment:
Correct me if I am wrong but this is missing in the updated code. We can set default values for attributes of a model!